Sources/sourcekit-lsp/SourceKitLSP.swift

```diff
@@ -169,6 +175,12 @@ struct SourceKitLSP: ParsableCommand {
   )
   var indexPrefixMappings = [PathPrefixMapping]()
 
+  @Option(
+    name: .customLong("workspace-type"),
+    help: "Override default workspace type selection; one of 'swiftpm', 'compdb', or 'buildserver'"
+  )
+  var workspaceType: WorkspaceType?
```

Comment:
I would prefer to name this `defaultWorkspaceType`, related to my comment that we should still fall back to other workspace types.